### PR TITLE
Update to 1.79

### PR DESCRIPTION
**Pull request by Veykril**
Builds on top of #499

### CODE REVIEW

File: src/program-structure-and-compilation.rst [hunk 1]
```diff
@@ -114,21 +114,30 @@ A :t:`crate` is a unit of compilation and linking that contains a tree of
...
 
 :dp:`fls_d9nn4yuiw1ja`
-A :t:`library crate` is either a :t:`crate` without a :t:`main function` or a
-:t:`crate` subject to :t:`attribute` :c:`no_main`. A tool is free to compile a
-:t:`library crate` to a shared library.
+A :t:`library crate` is either a :t:`crate` that does not have a :t:`function`
+in scope of its :t:`crate root module` under the :t:`name` ``main`` with a
+:t:`main function signature` or a :t:`crate` subject to :t:`attribute`
+:c:`no_main`. A tool is free to compile a :t:`library crate` to a shared
+library.
```
Comment (pietroalbini) [inline src/program-structure-and-compilation.rst:131]: I am quite confused by this: whether a crate is a binary crate or a library crate is just determined by the `--crate-type` flag or the `#![crate_type]` attribute. You can also have executables with `#![no_main]` (if you define your own entry point the linker can find).

**Review by pietroalbini — CHANGES_REQUESTED** [review]

File: src/functions.rst [hunk 2]
```diff
@@ -176,19 +176,20 @@ An :t:`unsafe function` is a :t:`function` subject to :t:`keyword` ``unsafe``.
 :dp:`fls_5hn8fkf7rcvz`
 The invocation of an :t:`unsafe function` shall require :t:`unsafe context`.
 
-:dp:`fls_nw49shkqx40b`
-A :t:`main function` is a :t:`function` that acts as an entry point into a
-program. A :t:`main function` is subject to the following restrictions:
+:dp:`fls_kNdIXl86g6Du`
+A :t:`function`'s :t:`name` shall not be the :t:`name` ``main`` unless it is in
+a :t:`crate root module` of a :t:`crate`.
```
[the remark this answers is not in the document]
Comment (Veykril) [reply src/functions.rst:181]: I have been tricked by the playground and me not reading the error message properly

**Review by pietroalbini — CHANGES_REQUESTED** [review]

File: src/attributes.rst [hunk 3]
```diff
@@ -1452,15 +1452,15 @@ Macros Attributes
 
 .. _fls_qyudjGHZfyJH:
 
-Attribute ``collapsable_debuginfo``
+Attribute ``collapse_debuginfo``
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (pietroalbini) [inline src/attributes.rst:1456]: Nit: the length of the header underline is not the same. I *think* it might cause Sphinx build to fail.

**Review by pietroalbini — APPROVED** [review]
bors merge

**Pull request merged**